src/Microsoft.Health.ControlPlane.Core/Features/Rbac/RbacService.cs

```diff
@@ -21,18 +22,28 @@ public RbacService(IControlPlaneDataStore controlPlaneDataStore)
             _controlPlaneDataStore = controlPlaneDataStore;
         }
 
+        public async Task DeleteIdentityProviderAsync(string name, string eTag, CancellationToken cancellationToken)
+        {
+            await _controlPlaneDataStore.DeleteIdentityProviderAsync(name, eTag, cancellationToken);
+        }
+
+        public async Task<IEnumerable<IdentityProvider>> GetAllIdentityProvidersAsync(CancellationToken cancellationToken)
+        {
+            return await _controlPlaneDataStore.GetAllIdentityProvidersAsync(cancellationToken);
+        }
+
         public async Task<IdentityProvider> GetIdentityProviderAsync(string name, CancellationToken cancellationToken)
         {
             EnsureArg.IsNotNullOrWhiteSpace(name, nameof(name));
 
             return await _controlPlaneDataStore.GetIdentityProviderAsync(name, cancellationToken);
         }
 
-        public async Task<IdentityProvider> UpsertIdentityProviderAsync(IdentityProvider identityProvider, CancellationToken cancellationToken)
+        public async Task<UpsertResponse<IdentityProvider>> UpsertIdentityProviderAsync(IdentityProvider identityProvider, string eTag, CancellationToken cancellationToken)
         {
             EnsureArg.IsNotNull(identityProvider, nameof(identityProvider));
 
-            return await _controlPlaneDataStore.UpsertIdentityProviderAsync(identityProvider, cancellationToken);
+            return await _controlPlaneDataStore.UpsertIdentityProviderAsync(identityProvider, eTag, cancellationToken);
```

Comment:
We should validate the identity provider before creating/updating. The validation should be similar to the role validation that currently exists. 

Besides the standard validation we should also verify that we can get to `{authority}/.well-known/openid-configuration` before saving.